test/test_parse_es.py

```diff
@@ -84,6 +86,134 @@ def test_extract_number_es(self):
+        self.assertEqual(extract_datetime(
+            "11 feb", lang='es', anchorDate=datetime(1998, 2, 1))[0],
+            datetime(1998, 2, 11))
+        self.assertEqual(extract_datetime(
+            "11 mar", lang='es', anchorDate=datetime(1998, 3, 1))[0],
+            datetime(1998, 3, 11))
+        self.assertEqual(extract_datetime(
+            "11 abr", lang='es', anchorDate=datetime(1998, 4, 1))[0],
+            datetime(1998, 4, 11))
+        self.assertEqual(extract_datetime(
+            "11 may", lang='es', anchorDate=datetime(1998, 5, 1))[0],
+            datetime(1998, 5, 11))
+        # there is an issue with the months of june through september (below)
+        # hay un problema con las meses junio hasta septiembre (lea abajo)
+        self.assertEqual(extract_datetime(
+            "11 oct", lang='es', anchorDate=datetime(1998, 10, 1))[0],
+            datetime(1998, 10, 11))
+        self.assertEqual(extract_datetime(
+            "11 nov", lang='es', anchorDate=datetime(1998, 11, 1))[0],
+            datetime(1998, 11, 11))
+        self.assertEqual(extract_datetime(
+            "11 dic", lang='es', anchorDate=datetime(1998, 12, 1))[0],
+            datetime(1998, 12, 11))
+
+        self.assertEqual(extract_datetime("", lang='es'), None)
+
+    # TODO fix bug causing these tests to fail (MycroftAI/mycroft-core#2348)
+    #         reparar error de traducción preveniendo las funciones abajo de
+    #         retornar correctamente
+    #         (escribido con disculpas por un Inglés hablante)
+    #      further broken tests are below their respective working tests.
+    @unittest.skip("currently processing these months incorrectly")
+    def test_bugged_output_wastebasket(self):
+        self.assertEqual(extract_datetime(
+            "11 jun", lang='es', anchorDate=datetime(1998, 6, 1))[0],
+            datetime(1998, 6, 11))
+        self.assertEqual(extract_datetime(
+            "11 junio", lang='es', anchorDate=datetime(1998, 6, 1))[0],
+            datetime(1998, 6, 11))
+        self.assertEqual(extract_datetime(
+            "11 jul", lang='es', anchorDate=datetime(1998, 7, 1))[0],
+            datetime(1998, 7, 11))
+        self.assertEqual(extract_datetime(
+            "11 ago", lang='es', anchorDate=datetime(1998, 8, 1))[0],
+            datetime(1998, 8, 11))
+        self.assertEqual(extract_datetime(
+            "11 sep", lang='es', anchorDate=datetime(1998, 9, 1))[0],
+            datetime(1998, 9, 11))
+
+        # It's also failing on years
+        self.assertEqual(extract_datetime(
+            "11 ago 1998", lang='es')[0], datetime(1998, 8, 11))
+
+    def test_extract_datetime_relative(self):
+        self.assertEqual(extract_datetime("esta noche", anchorDate=datetime(
+            1998, 1, 1), lang='es'),
+            [datetime(1998, 1, 1, 21, 0, 0), 'esta'])
+        self.assertEqual(extract_datetime(
+            "ayer noche", anchorDate=datetime(1998, 1, 1),
```

Comment:
Yet "ayer por la noche" or "ayer noche" aren't wrong at all, we rather use "anoche" to specify "last night".

Reply:
Sorry, I keep forgetting to check all of these. So far, this is the only one where I don't defer to you ;)

This particular test was to ensure that "ayer" and "noche" combined gets yesterday and evening, rather than returning either tonight or yesterday morning, as you'd get if you passed it "ayer" or "noche" separately.

That said, if there isn't one, there should be a test for "anoche" That's an oversight on my part.

Reply:
Adding a test for _anoche_ (to work exactly like ayer noche) does not pass. Let's create an issue for this after the PR is merged.